app/models/patient.rb

```diff
@@ -172,7 +172,7 @@ def last_arv_drug_expire_date
       order by orders.auto_expire_date desc
       limit 1
     SQL
-    result['auto_expire_date']&.to_date || nil
+    result['auto_expire_date']&.to_date || nil if !result.blank?
```

Comment:
For readability
```suggestion
result['auto_expire_date']&.to_date || nil if result.present?
```
OR

```suggestion
result['auto_expire_date']&.to_date || nil unless result.blank?
```